### PR TITLE
Add section dropdown

### PR DESCRIPTION
**Pull request by maddiedierker**
**Note: Change base back to `staging` before merging!**

This adds the section dropdown to the new dashboard-served teacher dashboard:

screen shot 2019-01-29 at 5 25 54 pm

### CODE REVIEW

File: apps/src/sites/studio/pages/teacher_dashboard/index.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ const baseUrl = `/teacher_dashboard/sections/${section.id}`;
 $(document).ready(function () {
   registerReducers({teacherSections, sectionData, manageStudents, sectionProgress, scriptSelection, stats, textResponses, sectionAssessments});
   const store = getStore();
+  store.dispatch(asyncSetSections());
```
Comment (islemaster) [inline apps/src/sites/studio/pages/teacher_dashboard/index.js]: Is there a reason we're getting the section data asynchronously from `/dashboardapi/sections` instead of including it in the initial view as part of `scriptData`, given we will always need it?

Comment (maddiedierker) [reply apps/src/sites/studio/pages/teacher_dashboard/index.js]: nope, that's a much better idea!

File: apps/src/templates/teacherDashboard/SelectSectionDropdown.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+class SelectSectionDropdown extends React.Component {
+  static propTypes = {
+    // Provided by redux.
+    sections: PropTypes.object,
+    selectedSectionId: PropTypes.number,
+  };
+
+  onChange = (event) => {
+    const sectionId = event.target.value;
+    navigateToHref(`/teacher_dashboard/sections/${sectionId}`);
```
Comment (davidsbailey) [inline apps/src/templates/teacherDashboard/SelectSectionDropdown.jsx:28]: We should consider eliminating this navigation at some point. 100% agree with keeping it "for now" since this gives us parity with pegasus teacher-dashboard and lets us move off angular soonest, though.

File: dashboard/app/controllers/teacher_dashboard_controller.rb [hunk 3]
```diff
@@ -1,6 +1,11 @@
 class TeacherDashboardController < ApplicationController
   def index
```
Comment (davidsbailey) [inline dashboard/app/controllers/teacher_dashboard_controller.rb:2]: the routes in this controller seem a bit non-standard -- I would recommend renaming `index` to `show` to more closely follow rails conventions. ok to do in a separate PR, although now might be a good time since you're adding a bunch of references to this route in the unit test.

**Review by davidsbailey — APPROVED** [review]
Excellent progress Maddie! One request to consider below.

**Review by islemaster — APPROVED** [review]
:tada: Yay this looks way cleaner to me.

**Comment by maddiedierker** [comment]
closing this PR, since it was a branch off of #26781, which i closed earlier. new PR is #26817, and i will address feedback there! this is a good lesson for me in not branching off of branches 😄
